cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/templates/etl/api/Emitter.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.templates.etl.api;
+
+/**
+ * Emit data.
+ *
+ * @param <O> Object that Emitter emits
+ */
+public interface Emitter<O> {
```

Comment:
I think the `Emitter` should have key and value.

```java
interface Emitter <K, V> {
  void emit(K key, V value);
}

interface ValueEmitter<V> extends Emitter<Void, V> {
  void emit(V value);
}
```

You can do similar thing in Transform. See how the Kafak flowlet is being done in the kafka-pack repo.